test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -868,17 +966,33 @@ void ProtoConverter::visit(ForStmt const& _x)
 {
 	bool wasInForBody = m_inForBodyScope;
 	bool wasInForInit = m_inForInitScope;
+	bool wasForInitScopeExtEnabled = m_forInitScopeExtEnabled;
 	m_inForBodyScope = false;
 	m_inForInitScope = true;
+	m_forInitScopeExtEnabled = true;
 	m_output << "for ";
 	visit(_x.for_init());
 	m_inForInitScope = false;
+	m_forInitScopeExtEnabled = wasForInitScopeExtEnabled;
 	visit(_x.for_cond());
 	visit(_x.for_post());
 	m_inForBodyScope = true;
 	visit(_x.for_body());
 	m_inForBodyScope = wasInForBody;
 	m_inForInitScope = wasInForInit;
+	if (m_inFunctionDef)
+	{
+		yulAssert(
+			!m_funcForLoopInitVars.empty() && !m_funcForLoopInitVars.back().empty(),
+			"Proto fuzzer: Invalid data structure");
+		// Remove variables in for-init
+		m_funcForLoopInitVars.back().pop_back();
+	}
+	else
+	{
+		yulAssert(!m_globalForLoopInitVars.empty(), "Proto fuzzer: Invalid data structure");
+		m_globalForLoopInitVars.pop_back();
+	}
```

Comment:
Depending on which scope we are in (global vs function), we pop back the list of for-init var decls before leaving the for loop. This is done because they should not be referencable beyond the for-loop